boot/bootutil/src/loader.c

```diff
@@ -965,7 +965,6 @@ boot_swap_image(struct boot_loader_state *state, struct boot_status *bs)
 }
 #endif
 
-#if (BOOT_IMAGE_NUMBER > 1)
```

Comment:
To pass the simulator you need to fix the non used function warning which happens in most build configurations. Enclose this with something like:

```
#if (BOOT_IMAGE_NUMBER > 1) || (MCUBOOT_OVERWRITE_ONLY && MCUBOOT_DOWNGRADE_PREVENTION)
```

Reply:
Should be fixed